UPGRADE-14.0.md

```diff
@@ -1440,6 +1440,17 @@ Follow the instructions in relevant sections, e.g. `shopsys/coding-standards` or
 -   replace search with SearchInput in search queries in Frontend API ([#3044](https://github.com/shopsys/shopsys/pull/3044))
     -   in `articlesSearch`, `brandSearch`, `categoriesSearch` and `productsSearch` queries, the `search` argument has been replaced with `searchInput` of type `SearchInput` update your usages of these queries appropriately
     -   see #project-base-diff to update your project
+-   fix not displaying hidden Adverts in frontend API ([#3065](https://github.com/shopsys/shopsys/pull/3065))
+    -   framework classes `AdvertController`, `AdvertFacade` and `AdvertRepository` have now set typehints and return typehints for all methods so you should update your extend project classes accordingly
+    -   method `Shopsys\FrameworkBundle\Model\Advert\AdvertRepository::getAdvertByPositionQueryBuilder()` has been renamed to `Shopsys\FrameworkBundle\Model\Advert\AdvertRepository::getVisibleAdvertByPositionQueryBuilder()` and its visibility has been changed from `protected` to `public`
+    -   constructor `Shopsys\FrontendApiBundle\Model\Advert\AdvertRepository::__constructor()` changed its interface:
+        ```diff
+            public function __construct(
+                protected readonly EntityManagerInterface $em,
+        +       protected readonly FrameworkAdvertRepository $advertRepository,
```

Comment:
```suggestion
        +       protected readonly FrameworkAdvertRepository $advertRepository, // use Shopsys\FrameworkBundle\Model\Advert\AdvertRepository as FrameworkAdvertRepository;
```

maybe this could help someone?